conans/client/remote_manager.py

```diff
@@ -160,8 +158,15 @@ def filter_function(urls):
             urls.pop(EXPORT_SOURCES_TGZ_NAME, None)
             return urls
 
-        zipped_files = self._call_remote(remote, "get_recipe", conan_reference, dest_folder,
-                                         filter_function)
+        t1 = time.time()
+        conan_reference, urls = self._call_remote(remote, "get_recipe_urls", conan_reference)
+        urls = filter_function(urls)
+        if not urls:
+            return conan_reference
+
+        dest_folder = self._client_cache.export(conan_reference)
+        zipped_files = self._call_remote(remote, "download_files_to_folder", urls, dest_folder)
+
```

Comment:
Previously these actions were done in the `rest_client` but we used that ugly filter_files and the ugliness was increasing, so now the rest_api:
a) gets urls
b) download urls

And is the remote manager who manages the file filtering calling first to a), then filtering then b)